client/components/tinymce/plugins/simple-payments/dialog/product-image-picker.jsx

```diff
@@ -0,0 +1,127 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Gridicon from 'gridicons';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+
+import MediaLibrarySelectedData from 'components/data/media-library-selected-data';
+import MediaModal from 'post-editor/media-modal';
+import MediaActions from 'lib/media/actions';
+import { getMediaItem } from 'state/selectors';
+import { localize } from 'i18n-calypso';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import ProductImage from 'components/tinymce/plugins/simple-payments/dialog/product-image';
+import RemoveButton from 'components/remove-button';
+
+class ProductImagePicker extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		featuredImage: PropTypes.object,
+		translate: PropTypes.func,
+	};
+
+	state = {
+		isSelecting: false,
+	};
+
+	showMediaModal = () => {
+		const { siteId, featuredImage } = this.props;
+
+		if ( featuredImage ) {
+			MediaActions.setLibrarySelectedItems( siteId, [ featuredImage ] );
+		}
+
+		this.setState( {
+			isSelecting: true,
+		} );
+	};
+
+	hideMediaModal = () => {
+		this.setState( {
+			isSelecting: false,
+		} );
+	};
+
+	setImage = value => {
+		this.hideMediaModal();
+
+		if ( ! value ) {
+			return;
+		}
+
+		this.props.input.onChange( value.items[ 0 ].ID );
+	};
+
+	removeCurrentImage = event => {
+		event.stopPropagation();
+
+		this.props.input.onChange( false );
+	};
+
+	getImagePlaceholder() {
+		return (
+			<div className="dialog__product-image-placeholder">
+				<Gridicon icon="add-image" size={ 36 } />
+				{ this.props.translate( 'Add an Image' ) }
+			</div>
+		);
+	}
+
+	getCurrentImage() {
+		const { siteId } = this.props;
+
+		return (
+			<div className="dialog__product-image">
+				<ProductImage siteId={ siteId } imageId={ this.props.input.value } />
+				<RemoveButton onRemove={ this.removeCurrentImage } />
+				<Gridicon icon="pencil" className="dialog__product-image-edit-icon" />
+			</div>
+		);
+	}
+
+	render() {
+		const { siteId, translate } = this.props;
+
+		if ( ! siteId ) {
+			return;
+		}
+
+		return (
+			<div>
+				<MediaLibrarySelectedData siteId={ siteId }>
+					<MediaModal
+						visible={ this.state.isSelecting }
+						onClose={ this.setImage }
+						siteId={ siteId }
+						labels={ { confirm: translate( 'Add' ) } }
+						enabledFilters={ [ 'images' ] }
+						single
+					/>
+				</MediaLibrarySelectedData>
+
+				<div className="dialog__pick-product-image-container" onClick={ this.showMediaModal }>
```

Comment:
Known a11y issue here - the clickable area should be a button/link but I needed a `div` here to preserve the original design. Will discuss alternatives with @iamtakashi.